static/docs/commands-reference/add.md

```diff
@@ -2,82 +2,193 @@
 3. Replace the file by a link to the file in the cache (see details below).
 4. Create a corresponding DVC file (metafile `.dvc`) and store the checksum
-to identify the right file in cache.
-
-Only _metafile_ (basically, pointer to the data in cache) is stored in Git,
-DVC manages data file contents.
-
-See [DVC File Format](/doc/user-guide/dvc-file-format) for the detailed
-description of the _metafile_ format.
-
-DVC stores the file's last modification timestamp, inode, and the checksum into
-a global state file `.dvc/state` to reduce time recomputing checksums later.
-
-Note, by default dvc tries a range of link types (reflink, hardlink, symlink,
-copy) to try to avoid copying any file contents and make dvc file operations
-very quick even for large files. Reflink is the best link type we could have,
-but even though it is becoming more and more common in modern filesystems, many
-filesystems still don't support it and thus dvc has to resort to a much more
-common hardlinks. See `dvc config` for more information.
+  to identify the cache entry.
+5. Add the _target_ filename to `.gitignore` to prevent it from being 
+  committed to the Git repository.
+6. Instructions are printed showing `git` commands for the files to be added to
+  the Git repository.
+
+The result is data file is added to the DVC cache, and the Git repository stores
+the metafile (`.dvc`).  The stage file (metafile) lists the added file as an
+`out` (output) of the stage, and references the DVC cache entry using the 
+checksum.  See [DVC File Format](/doc/user-guide/dvc-file-format) for the 
+detailed description of the DVC _metafile_ format.
+
+By default DVC tries a range of link types (`reflink`, `hardlink`, `symlink`,
+or `copy`) to try to avoid copying any file contents and to optimize DVC file 
+operations even for large files. The `reflink` is the best link type available, 
+but even though it is frequently supported by modern filesystems, many others 
+still don't support it.  DVC has the other link types for use on filesystems 
+without `reflink` support. See `dvc config` for more information.
+
+A `dvc add` target can be an individual file or a directory.  There are two ways
+to work with directory hierarchies with `dvc add`.
+
+1. With `dvc add --recursive`, the hierarchy is traversed and every file is 
+  added individually as described above.  This means every file has its own 
+  `.dvc` file, and a corresponding DVC cache entry is made.  If the 
+  `--no-commit` flag is added the DVC cache entry is not made.
+2. When not using `--recursive` a DVC stage file is created for the top of 
+  the directory (`dirname.dvc`), and every file in the hierarchy is added to the 
+  DVC cache, but these files do not have individual DVC files.  Instead the DVC 
+  file for the directory has a corresponding file in the DVC cache containing 
+  references to the files in the directory hierarchy.
+
+The `dvc add` command is useful for manually maintaining updates to files.  For
```

Comment:
to datasets and files ... and input datasets . `dvc add` is an essential part if you built pipelines with `dvc run`. One has to use it to add input data.